File: natura_app/lib/Domain/ApiRequestModel.dart
class ApiRequestModel {
  final AuthApiModel auth;
  final List<CustomHeaderModel> headers;
  final String? typeRequest;
  final String? url;
  final String? body;

  ApiRequestModel(
      {required this.auth,
        required this.headers,
        this.typeRequest,
        this.url,
        this.body});
}

class AuthApiModel {
  final String? type;
  final String? authorization;

  AuthApiModel({this.type, this.authorization});
}

class CustomHeaderModel {
  final String? header;
  final String? value;

  CustomHeaderModel({this.header, this.value});
}

File: API_Register/Domain/Models/Settings/ApiRequestModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Models.Settings
{
    public class ApiRequestModel
    {
        public string? Url { get; set; }
        public string TypeRequest { get; set; }
        public string? Body { get; set; }
        public List<CustomHeaderModel>? Headers { get; set; }
        public AuthApiModel? Auth { get; set; }
    }
    public class AuthApiModel
    {
        public string? Type { get; set; }
        public string? Authorization { get; set; }
    }
    public class CustomHeaderModel
    {

        public string? Header { get; set; }
        public string? Value { get; set; }
    }
}
